u8timeseries/models/statistics.py

```diff
@@ -0,0 +1,198 @@
+
+    gradient = np.gradient(r)
+    gradient_signs_changes = np.diff(np.sign(gradient))
+
+    if m is None:
+        # Tries to infer seasonality from AutoCorrelation Function if no value of m has been provided.
+        # We look for the first positive significant local maximum of the ACF by checking the sign changes
+        # in the gradient.
+
+        # Local maximum is indicated by signs_change == -2.
+        if len(np.nonzero((gradient_signs_changes == -2))[0]) == 0:
+            return False, 0
+
+        candidate = np.nonzero((gradient_signs_changes == -2))[0].tolist()
+
+    else:
+        # Check for local maximum when m is user defined.
+        if m - 1 not in np.nonzero((gradient_signs_changes == -2))[0]:
+            return False, m
+
+        candidate = [m - 1]
+
+    # Remove r[0], the auto-correlation at lag order 0, that introduces bias.
+    r = r[1:]
+
+    # The non-adjusted upper limit of the significance interval.
+    band_upper = r.mean() + norm.ppf(1 - alpha/2)*r.var()
+
+    # Significance test, stops at first admissible value.
+    is_significant = False
+    i = 0
+    c = 0
+
+    while not is_significant and i < len(candidate):
+        c = candidate[i] + 1
+        stat = _bartlett_formula(r, c - 1, len(ts))
+
+        is_significant = r[c-1] > stat * band_upper
+        i += 1
+
+    return is_significant, c
+
+
+def _bartlett_formula(r, m, length) -> float:
+    """
+    Computes the standard error of [r] at order [m] with respect to [length] according to Bartlett's formula.
+
+    :param r: The array whose standard error is to be computed.
+    :param m: The order of the standard error.
+    :param length: The size of the underlying sample to be used.
+    :return: The standard error of [r] with order [m].
+    """
+    if m == 1:
+        return math.sqrt(1/length)
+    else:
+        return math.sqrt((1 + 2 * sum(map(lambda x: x ** 2, r[:m-1]))) / length)
+
+
+def extract_trend_and_seasonality(ts: 'TimeSeries', freq: int = None, model: str = 'multiplicative') \
+                                                                -> Tuple['TimeSeries', 'TimeSeries']:
+    """
+    Extracts trend and seasonality from the TimeSeries [ts] using statsmodels.seasonal_decompose.
+
+    :param ts: The TimeSeries to study.
+    :param freq: The seasonality period to use.
+    :param model: The type of decomposition to use ('additive' or 'multiplicative').
+    :return: A tuple (trend, seasonal) of TimeSeries containing the trend and seasonality respectively.
+    """
+
+    decomp = seasonal_decompose(ts.pd_series(), freq=freq, model=model)
+
+    season = TimeSeries.from_times_and_values(ts.time_index(), decomp.seasonal)
+    trend = TimeSeries.from_times_and_values(ts.time_index(), decomp.trend)
+
+    return trend, season
+
+
+def remove_from_series(ts: 'TimeSeries', other: 'TimeSeries', model: str) -> 'TimeSeries':
```

Comment:
How about renaming `model` to `method` ?

Reply:
Not sure about that. The terminology comes from choosing either an additive or multiplicative decomposition of your time series  (the "model"). Also model is compatible with statstools.

Reply:
Indeed, it is not the decomposition that use the addition or the multiplication it is dependent on the model we use for the timeseries (see notes at the end of the `seasonal_decompose` doc [here](https://www.statsmodels.org/dev/generated/statsmodels.tsa.seasonal.seasonal_decompose.html))

Reply:
Oh ok my bad